ob1k-core/src/main/java/com/outbrain/ob1k/client/dispatch/DispatchStrategy.java

```diff
@@ -0,0 +1,15 @@
+package com.outbrain.ob1k.client.dispatch;
+
+import com.outbrain.ob1k.client.endpoints.DispatchAction;
+import com.outbrain.ob1k.client.targets.TargetProvider;
+import com.outbrain.ob1k.concurrent.ComposableFuture;
+import rx.Observable;
+
+/**
+ * @author marenzon
+ */
+public interface DispatchStrategy {
+
+  ComposableFuture dispatchAsync(TargetProvider targetProvider, DispatchAction<ComposableFuture> dispatchAction);
+  Observable dispatchStream(TargetProvider targetProvider, DispatchAction<Observable> dispatchAction);
```

Comment:
As all implementations of dispatchStream look the same maybe it can be extracted as a default method of the interface